exec.go

```diff
@@ -848,14 +848,20 @@ func runwaittask(c *Command, cpid chan *os.Process) {
 }
 
 // fsopenfd opens a plan9 Fid and makes a Go os.File from it for path on fsys.
-func fsopenfd(fsys *client.Fsys, path string, mode uint8) (*os.File, *client.Fid) {
+// Caller must close the returned file. This function will close the
+// other end of the pipe and the plan9 Fid.
+func fsopenfd(fsys *client.Fsys, path string, mode uint8) *os.File {
```

Comment:
Where do the `fid` instances get closed? My understanding from #74 is that the child Edwood must close the `fid` to the parent explicitly so that the `xfid.go:/xfidclose`'s `QWwrsel` doesn't change the selection at some inconvenient time. This change would seem to have altered this?

Reply:
`fid` gets closed inside the two loops in `fsopenfd`. Before, it was being closed twice: once inside `fsopenfd` and once in `Closeall`. I removed the close from `Closeall` because it's not possible for us to know if we're done writing to `fid` from that function. Writes are being buffered in the pipe. The goroutine inside `fsopenfd` closes the `fid` when it knows it's done reading from the pipe and writing to `fid`.

This is at least an improvement from before. Now writes won't get truncated because `fid` is closed by `Closeall` before writes to `fid` are done. The real fix is of course to not buffer writes in a pipe and pass `fid` directly to `exec.Cmd`. Then once `cmd.Wait` returns, we know that writes to `fid` are done and we can close the `fid` in `Closeall`. I added a TODO for this and will send a separate PR for it.

Reply:
This is what I have in mind: https://github.com/fhs/edwood/pull/1
(github isn't making it easy to send dependent PRs)

Reply:
I agree with your statement "`fid` gets closed...". What I don't understand is how the bug in #74 would then be possible and it (empirically) was definitely a result of the fid being closed much later. You tested this case? It's reasonably easy to replicate.